File: BookingApp/src/main/java/com/example/BookingApp/users/model/Complaint.java
package com.example.BookingApp.users.model;

import com.example.BookingApp.renting.model.RentingItem;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Complaint {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String description;
    @ManyToOne(fetch = FetchType.LAZY,cascade=CascadeType.ALL)
    private Client sender;
    @OneToOne(fetch = FetchType.LAZY,cascade=CascadeType.ALL)
    private RentingItem rentingItem;
}

File: BookingApp/frontend/src/app/shared/models/Complaint.ts
import { Client } from "./Client";
import { RentingItem } from "./reservations/RentingItem";

export class Complaint{
    public id : Number;
    public clientId : Number;
    public rentingItemId : Number;
    public description : String;
    public client : Client;
    public rentingItem : RentingItem;

    constructor(id:Number,clientId:Number,rentingItemId:Number,description:String){
        this.id = id;
        this.clientId = clientId;
        this.rentingItemId = rentingItemId;
        this.description = description;
    }
}